packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -265,10 +284,19 @@ export function createResponseState(
         typeof scriptConfig === 'string' ? scriptConfig : scriptConfig.src;
       const integrity =
         typeof scriptConfig === 'string' ? undefined : scriptConfig.integrity;
-      bootstrapChunks.push(
-        startModuleSrc,
-        stringToChunk(escapeTextForBrowser(src)),
-      );
+      if (nonce === undefined) {
+        bootstrapChunks.push(
+          startModuleSrc,
+          stringToChunk(escapeTextForBrowser(src)),
+        );
+      } else {
+        bootstrapChunks.push(
+          startModuleSrcWithNonce,
+          stringToChunk(escapeTextForBrowser(nonce)),
+          middleModuleSrcWithNonce,
+          stringToChunk(escapeTextForBrowser(src)),
+        );
+      }
```

Comment:
I would maybe just do this like the integrity attribute
```suggestion
      bootstrapChunks.push(
        startModuleSrc,
        stringToChunk(escapeTextForBrowser(src))
      );
      if (nonce) {
        bootstrapChunks.push(
          scriptNonce,
          stringToChunk(escapeTextForBrowser(nonce)),
        );
      }
```